examples/vae_text/vae_train.py

```diff
@@ -107,14 +103,13 @@ def __init__(self,
                 input_size=(self.decoder_w_embedder.dim +
                     config_model.batch_size),
                 hparams={"rnn_cell": config_model.dec_cell_hparams})
-            decoder_initial_state_size = (self.decoder.cell.hidden_size,
-                                          self.decoder.cell.hidden_size)
+            decoder_initial_state_size = (self.decoder.cell.hidden_size,) * 2
             sum_state_size = sum(decoder_initial_state_size)
 
         elif config_model.decoder_type == 'transformer':
             decoder_initial_state_size = torch.Size(
                 [1, config_model.dec_emb_hparams["dim"]])
-            sum_state_size = np.prod(decoder_initial_state_size)
+            sum_state_size = config_model.dec_emb_hparams["dim"]
             # position embedding
             self.decoder_p_embedder = tx.modules.SinusoidsPositionEmbedder(
                 position_size=config_model.max_pos,
```

Comment:
And also for line 118. You should create a function that generates word embeddings combined with positional embeddings, and pass that function as `token_pos_embedder`. I would suggest making that function a method of this `VAE` class.

Reply:
And in line 123, instead of raising `NotImplementedError`, let's use `ValueError` with an error message. `NotImplementedError` has a very specific use case: for abstract methods that ought to be overridden by subclasses.